File: 11-6-25_Challenge88/weekdayFinder.py
from datetime import datetime

def get_weekday(date_string):

    weekdayList = ["Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday"]

    dateObj = datetime.strptime(date_string, "%Y-%m-%d").date()
    weekday = weekdayList[dateObj.weekday()]

    return weekday

print(get_weekday("2025-11-06"))

File: 11-6-25_Challenge88/weekdayFinder.js
function getWeekday(dateString) {
    // getDay function returns a number so I can match that to the index to get the actual day
    const weekdayList = ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"];

    // the dates in the test are written with "-" so I need to replace them with "/" 
    // so the timezone doesn't get parsed wrong and give me one day earlier than I wanted.
    // https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off
    const date = new Date(dateString.replace(/-/g, '\/'));

    let weekday = weekdayList[date.getDay()];

    return weekday;
}

getWeekday("2025-11-06")